streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextUtils.java

```diff
@@ -47,9 +48,42 @@ public static StreamsMetricsImpl getMetricsImpl(final ProcessorContext context)
         return (StreamsMetricsImpl) context.metrics();
     }
 
+    /**
+     * Should be removed as part of KAFKA-10217
+     */
+    public static StreamsMetricsImpl getMetricsImpl(final StateStoreContext context) {
+        return (StreamsMetricsImpl) context.metrics();
+    }
```

Comment:
Added a bunch of duplicated extractors here to help keep the implementation classes clean.